cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -52,11 +67,47 @@ public HttpExecutorImpl(HttpClient client) {
         this.httpClient = client;
     }
 
+    public Organization getOrganization(URI uri) throws ApiGatewayException {
+        return fromSubSubunit(getSubSubUnitDtoWithMultipleEfforts(uri));
+    }
+
+    protected SearchResponse<Organization> query(URI uri) throws NotFoundException, FailedHttpRequestException {
+        HttpResponse<String> response = sendRequestMultipleTimes(addLanguage(uri)).get();
+        if (isSuccessful(response.statusCode())) {
+            try {
+                List<Organization> organizations = getOrganizations(response);
+                return new SearchResponse<Organization>(uri)
+                        .withHits(organizations)
+                        .withSize(getCount(response, organizations));
+            } catch (JsonProcessingException e) {
+                throw new FailedHttpRequestException(e.getMessage());
+            }
+        } else if (response.statusCode() == HTTP_NOT_FOUND) {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, uri));
+        } else {
+            throw new FailedHttpRequestException(errorMessage(response));
+        }
+    }
+
+
+    private Organization getParentOrganization(SubSubUnitDto subSubUnitDto) {
+        URI parent = Optional.ofNullable(subSubUnitDto.getParentUnit()).map(InstitutionDto::getUri).orElse(null);
+        final Set<Organization> partOf = isNull(parent)
+                ? null
+                : Set.of(getParentOrganization(attempt(() ->
+                getSubSubUnitDtoWithMultipleEfforts(parent)).orElseThrow()));
+        return new Organization.Builder()
+                .withId(getNvaApiId(subSubUnitDto.getId(), ORGANIZATION_PATH))
+                .withPartOf(partOf)
+                .withName(subSubUnitDto.getUnitName()).build();
+    }
+
     private Organization fromSubSubunit(SubSubUnitDto subSubUnitDto) {
```

Comment:
Very similar to `getParentOrganization` . Might not be a problem

Reply:
It is a tiny difference in getting the siblings children - they are not so interesting when going upwards in the tree